### DISCUSSION TITLE
Separate a SurfaceMesh into connected components

### DISCUSSION THREAD

**Discussion by chaohuang** [pmp-library/pmp-library#162] [General]
Given a `SurfaceMesh` with multiple connected components, what's the simplest way to separate it into multiple smaller `SurfaceMesh`, each of which corresponds to one connected component?

Thanks.

**Comment by dsieger** [comment]
We currently don't have that functionality, but it should be easy to implement using a depth first search on the mesh. Actually this would be a welcome contribution!

**Comment by dsieger** [comment]
FYI I added a simple function for detecting connected components in [0246bbc](https://github.com/pmp-library/pmp-library/commit/0246bbc09cb484fce14e7b337cd32d1bab6d1ee8). You can split the mesh based on the `v:component` vertex property added to the mesh.
